src/test/java/seedu/hirehub/logic/commands/SearchApplicationCommandTest.java

```diff
@@ -0,0 +1,242 @@
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.singletonList(ALICE_BACK_END), model.getFilteredApplicationList());
+
+        expectedMessage = String.format(MESSAGE_APPLICATIONS_LISTED_OVERVIEW, 0);
+        SearchPredicate<Application> thirdPersonDescriptor = new SearchApplicationPredicateBuilder()
+                .withPerson(nonExistingPerson).build();
+        command = new SearchApplicationCommand(thirdPersonDescriptor);
+        expectedModel.updateFilteredApplicationList(x -> thirdPersonDescriptor.test(x));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredApplicationList());
+    }
+
+    /**
+     * Test search_application method for searching a matching status in an uniqueFilteredApplicationList
+     */
+    @Test
+    public void execute_matchStatus() {
+
+        Status nonExistingStatus = new Status("REJECTED");
+
+        model = expectedModel;
+
+        String expectedMessage = String.format(MESSAGE_APPLICATIONS_LISTED_OVERVIEW, 2);
+        SearchPredicate<Application> firstStatusDescriptor = new SearchApplicationPredicateBuilder()
+                .withStatus(new Status("OFFERED")).build();
+        SearchApplicationCommand command = new SearchApplicationCommand(firstStatusDescriptor);
+        expectedModel.updateFilteredApplicationList(x -> firstStatusDescriptor.test(x));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Arrays.asList(BENSON_BACK_END, DANIEL_QUANTITATIVE_RESEARCHER),
+                model.getFilteredApplicationList());
+
+        expectedMessage = String.format(MESSAGE_APPLICATIONS_LISTED_OVERVIEW, 0);
+        SearchPredicate<Application> secondStatusDescriptor = new SearchApplicationPredicateBuilder()
+                .withStatus(nonExistingStatus).build();
+        command = new SearchApplicationCommand(secondStatusDescriptor);
+        expectedModel.updateFilteredApplicationList(x -> secondStatusDescriptor.test(x));
+        assertCommandSuccess(command, model, expectedMessage, expectedModel);
+        assertEquals(Collections.emptyList(), model.getFilteredApplicationList());
+    }
+
+    /**
+     * Test for search_application method given multiple searching criteria
+     */
+    @Test
+    public void execute_matchMultipleFields() {
+
+        Application testApplication = new ApplicationBuilder().withStatus(new Status("OFFERED"))
+                .withJob(QUANTITATIVE_RESEARCHER).build();
+
+        expectedModel.addApplication(testApplication);
+
+        model = expectedModel;
+
+        String expectedMessage = String.format(MESSAGE_APPLICATIONS_LISTED_OVERVIEW, 1);
+        SearchPredicate<Application> firstMultipleFieldPredicate = new SearchApplicationPredicateBuilder()
+                .withJob(FRONT_END).withPerson(CARL).withStatus(new Status("WAITLIST")).build();
```

Comment:
(Nit) Should the order be Person > Job > Status (also in the ApplicationPredicate and Application builder classes)? It is ok to ignore this if the effort required is too much.

Reply:
Sure, I will make modification.